var/spack/repos/builtin/packages/python/package.py

```diff
@@ -1132,82 +870,306 @@ def setup_dependent_build_environment(self, env, dependent_spec):
+
+        Note: This function really only makes
+        sense to use on Windows, could be overridden to
+        cross compile however.
+        """
+
+        arch_map = {"AMD64": "x64", "x86": "Win32", "IA64": "Win32", "EM64T": "Win32"}
+        arch = platform.machine()
+        if arch in arch_map:
+            arch = arch_map[arch]
+        return arch
+
+    @property
+    def win_build_params(self):
+        """
+        Arguments must be passed to the Python build batch script
+        in order to configure it to spec and system.
+        A number of these toggle optional MSBuild Projects
+        directly corresponding to the python support of the same
+        name.
+        """
+        args = []
+        args.append("-p %s" % self.plat_arch)
+        if self.spec.satisfies("+debug"):
+            args.append("-d")
+        if self.spec.satisfies("~ctypes"):
+            args.append("--no-ctypes")
+        if self.spec.satisfies("~ssl"):
+            args.append("--no-ssl")
+        if self.spec.satisfies("~tkinter"):
+            args.append("--no-tkinter")
+        return args
+
+    def build(self, pkg, spec, prefix):
+        # Windows builds use a batch script to drive
+        # configure and build in one step
+        with working_dir(self.stage.source_path):
+            pcbuild_root = os.path.join(self.stage.source_path, "PCbuild")
+            builder_cmd = os.path.join(pcbuild_root, "build.bat")
+            try:
+                subprocess.check_output(  # novermin
+                    " ".join([builder_cmd] + self.win_build_params), stderr=subprocess.STDOUT
+                )
+            except subprocess.CalledProcessError as e:
+                raise ProcessError(
+                    "Process exited with status %d" % e.returncode,
+                    long_message=e.output.decode("utf-8"),
+                )
+
+    def install(self, pkg, spec, prefix):
+        """
+        Python on Windows does not export an install target
+        so we must handcraft one here. This structure
+        directly mimics the install tree of the Python
+        Installer on Windows.
+
+        Parameters:
+            prefix (str): Install prefix for package
+        """
+        proj_root = self.stage.source_path
+        pcbuild_root = os.path.join(proj_root, "PCbuild")
+        build_root = os.path.join(pcbuild_root, platform.machine().lower())
+        include_dir = os.path.join(proj_root, "Include")
+        copy_tree(include_dir, prefix.include)
+        doc_dir = os.path.join(proj_root, "Doc")
+        copy_tree(doc_dir, prefix.Doc)
+        tools_dir = os.path.join(proj_root, "Tools")
+        copy_tree(tools_dir, prefix.Tools)
+        lib_dir = os.path.join(proj_root, "Lib")
+        copy_tree(lib_dir, prefix.Lib)
+        pyconfig = os.path.join(proj_root, "PC", "pyconfig.h")
+        copy(pyconfig, prefix.include)
+        shared_libraries = []
+        shared_libraries.extend(glob.glob("%s\\*.exe" % build_root))
```

Comment:
```suggestion
        shared_libraries.extend(glob.glob(r"%s\*.exe" % build_root))
```
If you make it a raw string you don't have to escape the directory path

Reply:
Addressed.